### PR TITLE
Long/hashing using bcrypt

### PR DESCRIPTION
**Pull request by khuolo01**
- Change hashing algorithm from _sha256_ to _bcrypt_
- Remove salt column from table users, for salt is already included in password_hash column by _bcrypt_
- Make some modifications for the tests based on the changes in User model.
- _test_user_signin_ uses hashing instead of mocking the hashing function.
- Create simple sign-in and sign-up templates.

### CODE REVIEW

File: message_app/user_sign_up.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ def api_user_signup():
             raise Exception(session["error"])
         else:
             # Hash the password
-            password, salt = hashing(password)
-            new_user = User(username=username, email=email, password_hash=password, password_salt=salt)
+            password_hash = hash_pw(password)
+            new_user = User(username=username, email=email, password_hash=password)
```
Comment (leth02) [inline message_app/user_sign_up.py]: Shouldn't this be `password_hash=password_hash`?

**Review by leth02 — CHANGES_REQUESTED** [review]
One minor question.

File: message_app/user_sign_up.py (hunk 1 above)
Comment (khuolo01) [reply message_app/user_sign_up.py]: yes

**Review by leth02 — APPROVED** [review]

**Comment by leth02** [comment]
LGTM

**Pull request merged**